tools/schema_registry_tools/schemas_command_generator.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# Usage:
+#
+# Say you have a dump file of the _schemas topic, taken via
+# 'rpk topic consume _schemas' and you want to restore it
+# directly to the _schemas topic again.
+#
+# Run the following command that parses the dump file and
+# generate runnable commands. This will fail without generating
+# the commands when a seq is higher than the corresponding offset.
+#
+# $ ./schemas_command_generator.py schemas-dump.txt > run.sh
+#
+# Here's how to restore the records:
+#
+# 1. Delete the existing _schemas topic
+# 2. Restart Redpanda
+# 3. Create the topic, configure the '-r' option based on the cluster size:
+#    $ rpk topic create _schemas -r 3 -c cleanup.policy=compact -c compression.type=none
+# 4. Run the generated script above
+#    $ chmod 755 run.sh
+#    $ sh run.sh
+#
+# Example outputs:
+# $ sh run.sh
+# Produced to partition 0 at offset 0 with timestamp 1662516019353.
+# Produced to partition 0 at offset 1 with timestamp 1662516020736.
+# Produced to partition 0 at offset 2 with timestamp 1662516021788.
+# Produced to partition 0 at offset 3 with timestamp 1662516022826.
+# Done
+#
+
+import json
+from pprint import pprint
+import re
+
+
+def main():
+    import argparse
+
+    def generate_options():
+        parser = argparse.ArgumentParser(
+            description='Redpanda Schema Registry _schemas Command Generator')
+        parser.add_argument('path', type=str, help='Path to the file')
+        return parser
+
+    parser = generate_options()
+    options, _ = parser.parse_known_args()
+
+    # Formatting as a single json
+    tmp = '['
+    with open(options.path) as f:
+        for l in f.read().splitlines():
+            if l.startswith('}'):
+                tmp += ('},')
+            else:
+                tmp += (l)
+    tmp = re.sub('},$', '}]', tmp)
+    j = json.loads(tmp)
+
+    # Generating rpk topic create commands
+    cmd_all = ''
+    cmd_all += f"#!/bin/bash\n\n"
+    for c, i in enumerate(j):
```

Comment:
We should enumerate all offsets between 0 and the largest offset.

For gaps in offsets (an offset where there is no record in the backup), we must create an innocuous record to "burn" the offset. I suggest a record with `"keytype": "NOOP"`.

Reply:
Got it.

Reply:
@BenPope , does inserting this make sense when there's a gap in offsets? Wonder if we should add `magic` or something others for the key.

```
echo '' | rpk topic produce _schemas --compression none -k '{"keytype":"NOOP"}'
```

Reply:
According to the code this looks sufficient.

Confluent appears to serialize with magic of 0, so probably worth doing for consistency: https://github.com/confluentinc/schema-registry/blob/7.2.0-post/core/src/main/java/io/confluent/kafka/schemaregistry/storage/NoopKey.java

Reply:
Followed by #6362